iXBRLViewerPlugin/iXBRLViewer.py

```diff
@@ -179,9 +180,16 @@ def addConcept(self, concept, dimensionType = None):
             if concept.isEnumeration:
                 conceptData["e"] = True
 
-            if concept.type.isTextBlock:
+            if concept.type is not None and concept.type.isTextBlock:
                 conceptData['t'] = True
 
+            if concept.isTypedDimension:
+                typedDomainElement = concept.typedDomainElement
+                if typedDomainElement is not None:
+                    typedDomainName = self.nsmap.qname(typedDomainElement.qname)
+                    conceptData['td'] = typedDomainName
```

Comment:
Thoughts on data backwards compatibility (i.e., old JSON, new viewer.js)?  I wonder if we should add a top-level data version key, and the viewer.js can disable features for older generated.

Reply:
For now I'm intentionally making sure we handle the unset state of new fields from the js side. The process of field removal is rather more complicated, considering that we allow folks to create plugins on top of the viewer.

It seems reasonable to me to version the JSON data alongside the project (as the JSON is akin to our library API for those extending the viewer). From a semver perspective, field removal or type change would necessitate a major version change, new fields would lead to minor changes, and other modifications would be deemed patches. Thoughts?

Reply:
> For now I'm intentionally making sure we handle the unset state of new fields from the js side.

Apologies, yes, I see that now.  I originally thought this was for `isTypedDimension() { return 'td' in this._c; }`, which would be wrong (absence of `td` means old JSON or non-typed-dimension), but that's not what the PR does.  (With that wrong assumption, I was trying to ask if we needed `if ('td' in fact._c) { typed } else if (dataVersion greater-than something) { explicit } else { unknown }`.)

> It seems reasonable to me to version the JSON data alongside the project (as the JSON is akin to our library API for those extending the viewer)

Agreed, we have [JavaScript Versioning](https://github.com/Workiva/ixbrl-viewer#javascript-versioning) section, and I think that's consistent with what you're saying.